src/number_format_base.tsx

```diff
@@ -191,17 +191,14 @@ export default function NumberFormatBase<BaseType = InputAttributes>(
 
   /**
    * if the formatted value is not synced to parent, or if the formatted value is different from last synced value sync it
-   * we also don't need to sync to the parent if no formatting is applied
```

Comment:
This behaviour contradicts what is described in the [documentation](https://s-yadav.github.io/react-number-format/docs/props/#onvaluechange-values-sourceinfo--):

>This handler provides access to any values changes in the input field and **is triggered only when a prop changes or the user input changes**

`value` is a prop, so any change should trigger the `onValueChange`. If the original intent of the function was to trigger on formatting changes, it should have been better described and/or be a different prop.